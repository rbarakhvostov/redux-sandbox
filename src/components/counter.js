import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';

import './counter.scss';

const Counter = ({ counter, inc, dec, rnd }) => {
  return (
    <div className="jumbotron">
      <h2>{ counter }</h2>
      <button
        id='dec'
        className="btn btn-primary btn-lg"
        onClick={dec}>
        DEC
      </button>
      <button
        id='inc'
        className="btn btn-primary btn-lg"
        onClick={inc}>
        INC
      </button>
      <button
        id='rnd'
        className="btn btn-primary btn-lg"
        onClick={rnd}>
        RND
      </button>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    counter: state,
  }
}

export { Counter };
export default connect(mapStateToProps, actions)(Counter);
