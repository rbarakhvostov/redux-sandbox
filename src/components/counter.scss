.jumbotron {
  display: flex;
  flex-direction: column;
  align-items: center;
    h2 {
      text-align: center;
      margin-bottom: 1em;
    }
    button:not(:last-child) {
      margin-bottom: 1em;
    }
}
