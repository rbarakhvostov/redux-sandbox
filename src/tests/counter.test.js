import React from 'react';
import { Counter } from '../components/counter';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Counter />', () => {
  it ('renders without crashing', () => {
    const wrapper = shallow(<Counter />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
