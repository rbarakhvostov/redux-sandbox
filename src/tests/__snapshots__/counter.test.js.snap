// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders without crashing 1`] = `
<div
  className="jumbotron"
>
  <h2 />
  <button
    className="btn btn-primary btn-lg"
    id="dec"
  >
    DEC
  </button>
  <button
    className="btn btn-primary btn-lg"
    id="inc"
  >
    INC
  </button>
  <button
    className="btn btn-primary btn-lg"
    id="rnd"
  >
    RND
  </button>
</div>
`;
